spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/ReactiveDatastoreTemplateTests.java

```diff
@@ -0,0 +1,542 @@
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the Reactive Datastore Template.
+ *
+ * @author Chengyuan Zhao
+ */
+public class ReactiveDatastoreTemplateTests extends AbstractDatastoreTemplateTests {
+
+	private ReactiveDatastoreTemplate reactiveDatastoreTemplate;
+
+	@Before
+	public void setup() {
+		super.setup();
+		this.reactiveDatastoreTemplate = new ReactiveDatastoreTemplate(this.datastoreTemplate);
+	}
+
+	public void setupGenericResultSets() {
+		when(this.datastore.run(any())).thenReturn(this.testEntityQueryResults);
+		when(this.datastore.get((Key[]) any())).thenReturn(Arrays.asList(this.e1, this.e2).iterator());
+		Iterator<Key> keys = Arrays.asList(this.key1, this.key2).iterator();
+
+		QueryResults<Key> queryResults = new QueryResults<Key>() {
+			@Override
+			public Class<?> getResultClass() {
+				return Key.class;
+			}
+
+			@Override
+			public Cursor getCursorAfter() {
+				return null;
+			}
+
+			@Override
+			public int getSkippedResults() {
+				return 0;
+			}
+
+			@Override
+			public QueryResultBatch.MoreResultsType getMoreResults() {
+				return null;
+			}
+
+			@Override
+			public boolean hasNext() {
+				return keys.hasNext();
+			}
+
+			@Override
+			public Key next() {
+				return keys.next();
+			}
+		};
+		when(this.datastore
+				.run(eq(Query.newKeyQueryBuilder().setKind("custom_test_kind").build())))
+						.thenReturn(queryResults);
+	}
+
+	@Test
+	public void findAllByIdTestNotNull() {
+		when(this.datastore.get((Key[]) any())).thenReturn(Collections.emptyIterator());
+		this.reactiveDatastoreTemplate
+				.findAllById(Collections.singletonList(this.badKey), AbstractDatastoreTemplateTests.TestEntity.class)
+				.collectList()
+				.subscribe(x -> assertThat(x).isEmpty());
+	}
+
+	@Test
+	public void findByIdTest() {
+		when(this.datastore.get((Key[]) any())).thenReturn(Arrays.asList(this.e1).iterator());
+		this.reactiveDatastoreTemplate.findById(this.key1, AbstractDatastoreTemplateTests.TestEntity.class)
+				.subscribe(result -> {
+					assertThat(result).isEqualTo(this.ob1);
+					assertThat(result.childEntities).contains(this.childEntity1);
+					assertThat(this.childEntity1).isEqualTo(result.singularReference);
+					assertThat(result.multipleReference).contains(this.childEntity1);
+				});
+	}
+
+	@Test
+	public void findByIdNotFoundTest() {
+		when(this.datastore.fetch(ArgumentMatchers.<Key[]>any())).thenReturn(Collections.singletonList(null));
+		verifyBeforeAndAfterEvents(null, new AfterFindByKeyEvent(Collections.emptyList(), Collections.singleton(null)),
+				() -> assertThat(this.datastoreTemplate.findById(createFakeKey("key0"),
+						AbstractDatastoreTemplateTests.TestEntity.class)).isNull(),
+				x -> {
+				});
+	}
+
+	@Test
+	public void findAllByIdTest() {
```

Comment:
is there a way to reuse code from the DatastoreTemplateTests that just swap in the template and then add the subscribe boilerplate.

Reply:
use .block() and you can just get the inner contents back.